src/sentry/incidents/models.py

```diff
@@ -238,3 +244,64 @@ class Meta:
         app_label = 'sentry'
         db_table = 'sentry_incidentsuspectcommit'
         unique_together = (('incident', 'commit'), )
+
+
+class AlertRuleStatus(Enum):
+    PENDING = 0
+    TRIGGERED = 1
+    PENDING_DELETION = 2
+    DELETION_IN_PROGRESS = 3
+
+
+class AlertRuleThresholdType(Enum):
+    ABOVE = 0
+    BELOW = 1
+
+
+class AlertRuleAggregations(Enum):
+    TOTAL = 0
+    UNIQUE_USERS = 1
+
+
+# TODO: This should probably live in the snuba app.
+class SnubaDatasets(Enum):
+    EVENTS = 'events'
+
+
+class AlertRuleManager(BaseManager):
+    """
+    A manager that excludes all rows that are pending deletion.
+    """
+
+    def get_queryset(self):
```

Comment:
@markstory I mentioned this the other day, we could potentially do something like this for groups and other objects that can be pending deletion. Obv a large piece of work, but might just help us avoid a class of bugs.

Reply:
I like this. Making it harder to make mistakes is :+1:. Should this manager also expose a method to get a queryset that can find deleted records?

Reply:
Yeah, definitely need that. I was going to implement that as part of the delete functionality, since at the moment that's the only logic that will need to access the deleted alert rule.